Forms/FormBuilder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * A service which can generate a form for a given record and controller
+ */
+interface FormBuilder {
+
+	/**
+	 * @return DataObject
+	 */
+	public function getRecord();
+
+	/**
+	 * @return Controller
+	 */
+	public function getController();
```

Comment:
Controller::curr() is not useful here, in that the form being built may be for one that is not the current controller. :)

The intention of exposing the controller in this way is to make it available to extensions. In extendAll() the current form builder is passed not only to extensions on itself, but also to the controller and the model. There may be times where extensions on the model may need to know which controller, and vice versa, extensions on the controller may need to know which model.